packages/client/utils/AtlassianManager.ts

```diff
@@ -831,4 +923,87 @@ export default abstract class AtlassianManager {
       `https://api.atlassian.com/ex/jira/${cloudId}/rest/api/3/screens/${screenId}/tabs/${tabId}/fields/${fieldId}`
     )
   }
+
+  async registerWebhooks(cloudId: string, webhooks: WebhookDetails[]) {
+    return this.post(`https://api.atlassian.com/ex/jira/${cloudId}/rest/api/3/webhook`, {
+      url: makeAppURL('http://localhost:3000', '/webhooks/jira'),
+      // url: makeAppURL(appOrigin, '/webhooks/jira'),
+      webhooks
+    })
+  }
+
+  async deleteWebhooks(cloudId: string, webhookIds: number[]) {
+    return this.delete(`https://api.atlassian.com/ex/jira/${cloudId}/rest/api/3/webhook`, {
+      webhookIds
+    })
+  }
+
+  async extendWebhooks(cloudId: string, webhookIds: number[]) {
+    return this.put(`https://api.atlassian.com/ex/jira/${cloudId}/rest/api/3/webhook/refresh`, {
+      webhookIds
+    })
+  }
+
+  async getWebhooks(cloudId: string) {
+    return this.get<JiraPageBean<Webhook>>(
+      `https://api.atlassian.com/ex/jira/${cloudId}/rest/api/3/webhook`
+    )
+  }
+  async getAllWebhooks(cloudIds: string[]) {
+    return Promise.all(cloudIds.map((cloudId) => this.getWebhooks(cloudId)))
+  }
+
+  async updateWebhook(webhookId: number, cloudId: string) {
+    // register new hook
+    await this.registerAllProjectsWebhooks([cloudId])
+    // delete old hook
+    await this.deleteWebhooks(cloudId, [webhookId])
+  }
+
+  async registerOrUpdateWebhooks(cloudIds: string[]) {
+    const webhooksAndErrors = await this.getAllWebhooks(cloudIds)
+    await Promise.all(
+      webhooksAndErrors.map(async (entry, index) => {
+        if (entry instanceof Error) {
+          return Promise.reject(Error)
+        } else if (entry instanceof RateLimitError) {
+          // retry
+          const waitTime = Date.now() - entry.retryAt.getUTCMilliseconds()
+          if (waitTime > 0) await sleep(waitTime)
+          return this.registerOrUpdateWebhooks([cloudIds[index]!])
+        } else if (entry.total > 0) {
+          return this.updateWebhook(entry.values[0]!.id, cloudIds[index]!)
+        } else {
+          return this.registerAllProjectsWebhooks([cloudIds[index]!])
+        }
+      })
+    )
+  }
+
+  async registerAllProjectsWebhooks(cloudIds: string[]) {
+    if (cloudIds.length === 0) return
+    const allProjects = await this.getAllProjects(cloudIds)
```

Comment:
-1 why not just watch all projects? by filtering on project, that means the webhook will go stale when they add a new project to a cloud

Reply:
Per my understanding (and i thought yours) its not possible to subscribe per cloud id, we actually discussed this [here](https://github.com/ParabolInc/parabol/issues/6153#issuecomment-1057517811) when we discussed the jql filter being required. Attempting to subscribe all was my first choice